Correct missing options from RCx_OPTIONS params and remove Winch from SERVOx_FUNCTION list

Hwurzburg opened the pull request:
while working on a common page for RC_Options, I found these were implemented but not listed in the code for the params

rmackay9 commented on libraries/RC_Channel/RC_Channel.cpp:
```diff
@@ -87,8 +87,8 @@ const AP_Param::GroupInfo RC_Channel::var_info[] = {
-    // @Values{Copter}: 0:Do Nothing, 2:Flip, 3:Simple Mode, 4:RTL, 5:Save Trim, 7:Save WP, 9:Camera Trigger, 10:RangeFinder, 11:Fence, 13:Super Simple Mode, 14:Acro Trainer, 15:Sprayer, 16:Auto, 17:AutoTune, 18:Land, 19:Gripper, 21:Parachute Enable, 22:Parachute Release, 23:Parachute 3pos, 24:Auto Mission Reset, 25:AttCon Feed Forward, 26:AttCon Accel Limits, 27:Retract Mount, 28:Relay On/Off, 3 ...
-    // @Values{Rover}: 0:Do Nothing, 4:RTL, 5:Save Trim, 7:Save WP, 9:Camera Trigger, 11:Fence, 16:Auto, 19:Gripper, 24:Auto Mission Reset, 28:Relay On/Off, 30:Lost Rover Sound, 31:Motor Emergency Stop, 34:Relay2 On/Off, 35:Relay3 On/Off, 36:Relay4 On/Off, 40:Proximity Avoidance, 41:ArmDisarm, 42:SmartRTL, 46:RC Override Enable, 50:LearnCruise, 51:Manual, 52:Acro, 53:Steering, 54:Hold, 55:Guided, ...
+    // @Values{Copter}: 0:Do Nothing, 2:Flip, 3:Simple Mode, 4:RTL, 5:Save Trim, 7:Save WP, 9:Camera Trigger, 10:RangeFinder, 11:Fence, 13:Super Simple Mode, 14:Acro Trainer, 15:Sprayer, 16:Auto, 17:AutoTune, 18:Land, 19:Gripper, 21:Parachute Enable, 22:Parachute Release, 23:Parachute 3pos, 24:Auto Mission Reset, 25:AttCon Feed Forward, 26:AttCon Accel Limits, 27:Retract Mount, 28:Relay On/Off, 3 ...
+    // @Values{Rover}: 0:Do Nothing, 3:Simple Mode, 4:RTL, 5:Save Trim, 7:Save WP, 9:Camera Trigger, 11:Fence, 16:Auto, 19:Gripper, 24:Auto Mission Reset, 28:Relay On/Off, 30:Lost Rover Sound, 31:Motor Emergency Stop, 34:Relay2 On/Off, 35:Relay3 On/Off, 36:Relay4 On/Off, 40:Proximity Avoidance, 41:ArmDisarm, 42:SmartRTL, 46:RC Override Enable, 50:LearnCruise, 51:Manual, 52:Acro, 53:Steering, 54:H ...
```
Looks like the Rover options have both "Simple Mode" and "Simple".  There's actually only one "Simple" mode in Rover though - not sure which of these two options it is.

For Copter it looks like 31 to 36 are out of place although it's an existing problem.

As a minor side note, I don't think we should have a speace between the number and the string.  Being picky but..

peterbarker replied:
I think this is because there's an actual Mode::Simple in Rover and there's the "enable pilot input transforms" option available in Copter.  They are distinct, and Copter should use one and Rover the other.  They definitely should not be merged - pilot input transforms would make sense for boats and omnis, for example.

Hwurzburg commented:
while trying to understand why object avoidance (40) enable is missing from Rover, I believe its also missing from Copter code....its not in Copter/RC_Channel.cpp...and the library default does nothing...wiki docs have it as a feature for some time in Copter, but it doesnt seem to be there....what am I missing?

peterbarker commented:
On Mon, 28 Oct 2019, Henry Wurzburg wrote:

> while trying to understand why object avoidance (40) enable is missing from Rover, I believe its also missing from Copter code....its not in
> Copter/RC_Channel.cpp...and the library default does nothing...wiki docs have it as a feature for some time in Copter, but it doesnt seem to be
> there....what am I missing?

The code in libraries/RC_Channel/RC_Channel.cpp?

Hwurzburg commented:
okay so both Rover and Copter have 40 and its active...thanks

Hwurzburg commented:
per Randy, removed Winch from RCx_OPTIONS....also removed Winch output from SERVOx_FUNCTIONS..will split and rename correctly tomorrow....too late tonight

Hwurzburg commented:
okay split the library commits into two for easy releases..summary of whats changed from master since reading it is difficult:
RC_Channel: re-odered Copter option sequences. removed Winch enable and control..added STANDBY mode..renamed Simple to SIMPLE mode in Rover..
SRV_Channel: removed winch option for output
